test/pytest/test.py

```diff
@@ -243,7 +242,29 @@ def testGetPartsOfValuesDocumentMultiple(self):
             self.assertOk(r.execute_command('JSON.SET', 'test',
                                             '.', json.dumps(docs['values'])))
             data = json.loads(r.execute_command('JSON.GET', 'test', *docs['values'].keys()))
-           # self.assertDictEqual(data, docs['values']) # TODO backward compatibility with JSONPATH "$.list" vs "list"
+            self.assertDictEqual(data, docs['values'])
+
+    def testGetFormatting(self):
+        with self.redis() as r:
+            r.client_setname(self._testMethodName)
+            r.flushdb()
+
+            d = {'dict': {'f': 'v'}}
+            a = {'arr': [0, 1]}
+            self.assertOk(r.execute_command('JSON.SET', 'd', '$', json.dumps(d)))
+            self.assertOk(r.execute_command('JSON.SET', 'a', '$', json.dumps(a)))
+
+            # No formatting
+            res = r.execute_command('JSON.GET', 'd')
+            self.assertEqual(res, json.dumps(d, separators=(",", ":")))
+            res = r.execute_command('JSON.GET', 'a')
+            self.assertEqual(res, json.dumps(a, separators=(",", ":")))
+
+            # All formatting options together
+            res = r.execute_command('JSON.GET', 'd', 'INDENT', '  ', 'NEWLINE', '\n', 'SPACE', ' ')
+            self.assertEqual(res, json.dumps(d, indent=2, separators=(",", ": ")))
+            res = r.execute_command('JSON.GET', 'a', 'INDENT', '  ', 'NEWLINE', '\n', 'SPACE', ' ')
```

Comment:
Maybe add a test where `SPACE` is `\t`?

Reply:
I thought of it but it's not not possible using python 2.7 `json.dumps()` (yet another reason to use python3).
I'll use custom code instead of python's json and make this much more thorough.